pkg/deploy/generator/resources_rp.go

```diff
@@ -1284,6 +1284,22 @@ func (g *generator) rpServiceKeyvaultAccessPolicies() []mgmtkeyvault.AccessPolic
 	}
 }
 
+func (g *generator) rpConfigKeyvaultAccessPolicies() []mgmtkeyvault.AccessPolicyEntry {
+	return []mgmtkeyvault.AccessPolicyEntry{
+		{
+			TenantID: &tenantUUIDHack,
+			ObjectID: to.StringPtr("[parameters('fpServicePrincipalId')]"),
```

Comment:
This should be the RP MSI, not the FP SP.  You're using the RP's MSI to setup a keyvault client above.